### PR TITLE
Fixed webapp response gzipping

### PR DESCRIPTION
**Pull request by Josca**

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -527,16 +527,16 @@ async def gzip_middleware(request, handler, **kwargs):
...
-        if not request.path.startswith('/api/v'):  # exception not to gzip swagger pages
+        try:
+            if res.body is not None and header in request.headers and "gzip" in request.headers[header]:
+                gzipped_body = gzip.compress(res.body, compresslevel=GZIP_COMPRESS_LEVEL)
+                res.body = gzipped_body
+                res.headers["Content-Encoding"] = "gzip"
+                return res
+        except (TypeError, AttributeError):
+            LOGGER.warning("unable to gzip response '%s'", request.path)
             return res
```
Comment (jdobes) [inline webapp/app.py:538]: Nothing is returned if condition on L531 is false. There are two returns in gzip_middleware function, I would keep one return at the end.

File: webapp/app.py [hunk 2]
```diff
@@ -527,16 +527,16 @@ async def gzip_middleware(request, handler, **kwargs):
         res = await handler(request, **kwargs)
         header = 'Accept-Encoding'
 
-        if not request.path.startswith('/api/v'):  # exception not to gzip swagger pages
+        try:
+            if res.body is not None and header in request.headers and "gzip" in request.headers[header]:
```
Comment (jdobes) [inline webapp/app.py:531]: Nothing is returned if condition on L531 is false. There are two returns in gzip_middleware function, I would keep one return at the end.

Comment (Josca) [reply webapp/app.py:531]: fixed

**Comment by jdobes** [comment]
updated commit message to contain semantic-release keyword

**Pull request merged**